pyccl/halos/halo_model.py

```diff
@@ -115,37 +116,35 @@ def _integ_spline(self, fM, lM):
 
     def _get_ingredients(self, a, cosmo, get_bf):
         # Compute mass function and bias (if needed) at a new
-        # value of the scale factor.
+        # value of the scale factor and/or with a new Cosmology
         rho0 = None
         if a != self._a_current_mf:
-            rho0 = cosmo.rho_x(1., "matter", is_comoving=True)
-            self.mf = self._massfunc.get_mass_function(cosmo, self._mass, a,
-                                                       mdef_other=self._mdef)
-            self.mf0 = (rho0 -
-                        self._integrator(self.mf * self._mass,
-                                         self._lmass)) / self._m0
             self._a_current_mf = a
+            rho0 = cosmo.rho_x(1., "matter", is_comoving=True)
+            self._mf = self._massfunc.get_mass_function(
+                cosmo, self._mass, a, mdef_other=self._mdef)
+            self._mf0 = (rho0 - self._integrator(
+                self._mf*self._mass, self._lmass)) / self._m0
 
         if get_bf:
+            self._a_current_mf = a
```

Comment:
why?

Reply:
yes I meant to put it under the if-statement, good that you noticed it